biz.aQute.bndlib/src/aQute/bnd/osgi/resource/ResourceBuilder.java

```diff
@@ -710,4 +710,170 @@ public boolean addFile(File file, URI uri) throws Exception {
 		addContentCapability(uri, sha256, file.length(), mime);
 		return hasIdentity;
 	}
+
+	public ResourceBuilder safeResourceBuilder() {
+		return new SafeResourceBuilder();
+	}
+
+	private static class SafeResourceBuilder extends ResourceBuilder {
```

Comment:
This cannot be static and have no reference to the real ResourceBuilder. :-)

I think you want to make this non-static, so it contain a reference to the real ResourceBuilder and replace the `super` calls with calls to `ResourceBuilder.this.`...